examples/raja-launch-mat-mat.cpp

```diff
@@ -0,0 +1,690 @@
+      printf("whoops!");
+    }
+  };
+
+
+  template<typename POLICY_LIST, typename SEGMENT, typename BODY>
+  RAJA_HOST_DEVICE
+  void loop(LaunchContext const &ctx, SEGMENT const &seg, BODY const &body){
+
+
+    LoopPlaceSwitchboard<POLICY_LIST, 0, camp::size<POLICY_LIST>::value>::exec(ctx, seg, body);
+
+
+  }
+
+} // namespace RAJA
+
+
+  using policy1_HOST = RAJA::LPolicy<RAJA::HOST, RAJA::loop_exec>;
+#ifdef RAJA_ENABLE_CUDA
+  using policy1x_DEVICE = RAJA::LPolicy<RAJA::DEVICE, RAJA::cuda_block_x_direct >;
+  using policy1y_DEVICE = RAJA::LPolicy<RAJA::DEVICE, RAJA::cuda_block_y_direct >;
+#else
+  using policy1_DEVICE = RAJA::LPolicy<RAJA::DEVICE, RAJA::loop_exec>;
+#endif
+  using outer0 = camp::list<policy1_HOST, policy1x_DEVICE>;
+  using outer1 = camp::list<policy1_HOST, policy1y_DEVICE>;
+
+  using policy2_HOST = RAJA::LPolicy<RAJA::HOST, RAJA::loop_exec>;
+#ifdef RAJA_ENABLE_CUDA
+  using policy2x_DEVICE = RAJA::LPolicy<RAJA::DEVICE, RAJA::cuda_thread_x_loop >;
+  using policy2y_DEVICE = RAJA::LPolicy<RAJA::DEVICE, RAJA::cuda_thread_y_loop >;
+#else
+  using policy2_DEVICE = RAJA::LPolicy<RAJA::DEVICE, RAJA::loop_exec>;
+#endif
+  using team0 = camp::list<policy2_HOST, policy2x_DEVICE>;
+  using team1 = camp::list<policy2_HOST, policy2y_DEVICE>;
+
+
+int main(){
+
+  //N is number of blocks in each matrix
+  const int NBlocks = 4;
+  const int NThreads = CUDA_BLOCK_SIZE;
+  const int N = CUDA_BLOCK_SIZE*NBlocks;
+
+//
+// Allocate and initialize matrix data.
+//
+  double *A = memoryManager::allocate<double>(N * N);
+  double *B = memoryManager::allocate<double>(N * N);
+  double *C = memoryManager::allocate<double>(N * N);
+
+  for (int row = 0; row < N; ++row) {
+    for (int col = 0; col < N; ++col) {
+      A(row, col) = row;
+      B(row, col) = col;
+    }
+  }
+
+  std::cout << "\n Running RAJA-Teams V2-version of matrix multiplication...\n";
+
+  for(int exec_place = 0;exec_place < 2;++ exec_place)
+  {
+    RAJA::ExecPlace select_cpu_or_gpu = (RAJA::ExecPlace)exec_place;
+    //auto select_cpu_or_gpu = RAJA::HOST;
+    //auto select_cpu_or_gpu = RAJA::DEVICE;
+
+    /*
+     * launch just starts a "kernel" it's doesn't provide any looping.
+     *
+     * The first argument determines which policy should be executed,
+     *
+     * The second argument is the number of teams+threads needed for each of the
+     * policies.
+     *
+     * Third argument is the lambda for the policy.
+     *
+     *
+     * The lambda takes a "resource" object, which has the teams+threads and
+     * policy selection information.
+     */
+
+    //Set up Teams/Threads
+
+    RAJA::launch(
+      select_cpu_or_gpu,
+      camp::make_tuple(
+        RAJA::Resources<RAJA::HOST>(RAJA::Threads(NBlocks, NBlocks)),
+        RAJA::Resources<RAJA::DEVICE>(RAJA::Teams(NBlocks, NBlocks), RAJA::Threads(NThreads, NThreads)) ),
+      [=] RAJA_HOST_DEVICE (RAJA::LaunchContext ctx)
+    {
```

Comment:
Working example of matrix matrix multiply with shared memory. Works with sequential and cuda backends.